Update Documentation & Readme to be current #74

morukele opened the pull request:
Hello @hgm-king, this commit has updated the readme.md and the lib.rs to reflect the TLS and Compression feature. 

It addresses the issue #74.

Let me know if you have any comments or changes you want me to make.

hgm-king reviewed (approved):
Great work, we appreciate the contribution :)

morukele commented:
@hgm-king, you are welcome. I am open to contributing to the project if you need more help. 

Feel free to tag me on any beginner or mid-level issue that I can work on.

hgm-king commented:
Actually, I have some tweaks to get the examples to compile correctly, let me share them in a second

hgm-king commented on README.md:
````diff
@@ -116,6 +118,111 @@ while let Some(batch) = stream.next().await {
+
+Example for Producer with TLS support:
+```rust
+let tls_option = TlsConnectionOptions {
````
On this one and all subsequent tlsConnection options, use something like:
```
let tls_option = samsa::prelude::TlsConnectionOptions {
        broker_options: vec![samsa::prelude::BrokerAddress {
            host: "127.0.0.1:9092".to_owned(),
            port: 9092,
        }],
        key: "/path_to_key_file".into(),
        cert: "/path_to_cert_file".into(),
        cafile: Some(("/path_to_ca_file").into()),
    };
```

morukele replied:
Okay, this makes sense. I hesitated to use the `.into()` method because I wanted to show that it was a BufPath. I will make the changed now

Another question: does this BrokerAddress class apply to the non-TLS variants?

hgm-king replied:
Yes, to make it consistent across both. Each connection type requires that BrokerAddress at a minimum

hgm-king commented on README.md:
```diff
@@ -116,6 +118,111 @@ while let Some(batch) = stream.next().await {
+        partition_id,
+        key: Some(bytes::Bytes::from_static(b"Tester")),
+        value: Some(bytes::Bytes::from_static(b"Value")),
+        headers: vec![String::from("Key"), bytes::Bytes::from("Value")]
```
The `Header` struct is not public to users of the crate :/ we forgot to export it from `lib.rs` so could you please add `pub use crate::protocol::Header;` to the exports in that file?

And then the usage here would be:
```
        headers: vec![samsa::prelude::Header::new(String::from("Key"), bytes::Bytes::from("Value"))],
```

morukele replied:
Yes I will

hgm-king commented on README.md:
```diff
@@ -116,6 +118,111 @@ while let Some(batch) = stream.next().await {
+        headers: vec![String::from("Key"), bytes::Bytes::from("Value")]
+    };
+
+let producer_client = samsa::prelude::ProducerBuilder::new(tls_option, vec![topic_name.to_string()])
```
the Producer and other Builders require a generic argument now, to specify which connection it uses.

This will work:
```
samsa::prelude::ProducerBuilder::<samsa::prelude::TlsConnection>::new
```

morukele replied:
Okay, this makes sense. I will make the changes

morukele commented:
@hgm-king, I have addressed all the issues in this new commit, including all the relevant changes. 

There is much less red in the examples in the `lib.rs` file now. Also, I had to change all instances of the topic from `&str` to `String`.

Let me know what you think.

hgm-king reviewed (approved):
Beautiful, thank you!
